receiver/k8sclusterreceiver/containers.go

```diff
@@ -0,0 +1,174 @@
+		" past, or > 0, in which case you can conclude there were restarts in the recent past, and" +
+		" not try and analyze the value beyond that.",
+	Unit: "1",
+	Type: metricspb.MetricDescriptor_GAUGE_INT64,
+}
+
+var containerReadyMetric = &metricspb.MetricDescriptor{
+	Name:        "kubernetes/container/ready",
+	Description: "Whether a container has passed its readiness probe (0 for no, 1 for yes)",
+	Type:        metricspb.MetricDescriptor_GAUGE_INT64,
+}
+
+// getStatusMetricsForContainer returns metrics about the status of the container.
+func getStatusMetricsForContainer(cs corev1.ContainerStatus) []*metricspb.Metric {
+	metrics := []*metricspb.Metric{
+		{
+			MetricDescriptor: containerRestartMetric,
+			Timeseries: []*metricspb.TimeSeries{
+				utils.GetInt64TimeSeries(int64(cs.RestartCount)),
+			},
+		},
+		{
+			MetricDescriptor: containerReadyMetric,
+			Timeseries: []*metricspb.TimeSeries{
+				utils.GetInt64TimeSeries(int64(boolToInt(cs.Ready))),
+			},
+		},
+	}
+
+	return metrics
+}
+
+func boolToInt(b bool) int {
+	if b {
+		return 1
+	}
+	return 0
+}
+
+var containerRequestMetric = &metricspb.MetricDescriptor{
+	Name:        "kubernetes/container/request",
+	Description: "Resource requested for the container",
+	Type:        metricspb.MetricDescriptor_GAUGE_INT64,
+	LabelKeys:   []*metricspb.LabelKey{{Key: "resource"}},
+}
+
+var containerLimitMetric = &metricspb.MetricDescriptor{
+	Name:        "kubernetes/container/limit",
+	Description: "Maximum resource limit set for the container",
+	Type:        metricspb.MetricDescriptor_GAUGE_INT64,
+	LabelKeys:   []*metricspb.LabelKey{{Key: "resource"}},
+}
+
+// getSpecMetricsForContainer metricizes values from the container spec.
+// This includes values like resource requests and limits.
+func getSpecMetricsForContainer(c corev1.Container) []*metricspb.Metric {
+	metrics := make([]*metricspb.Metric, 0)
+
+	for _, t := range []struct {
+		metric *metricspb.MetricDescriptor
+		rl     corev1.ResourceList
+	}{
+		{
+			containerRequestMetric,
+			c.Resources.Requests,
+		},
+		{
+			containerLimitMetric,
+			c.Resources.Limits,
+		},
+	} {
+		for k, v := range t.rl {
+			val := v.Value()
+			if k == corev1.ResourceCPU {
+				val = v.MilliValue()
+			}
+
+			metrics = append(metrics,
+				&metricspb.Metric{
+					MetricDescriptor: t.metric,
+					Timeseries: []*metricspb.TimeSeries{
+						utils.GetInt64TimeSeriesWithLabels(val, []*metricspb.LabelValue{{Value: string(k)}}),
+					},
+				},
+			)
+		}
+	}
+
+	return metrics
+}
+
+// getResourceForContainer returns a proto representation of the pod.
+func getResourceForContainer(labels map[string]string) *resourcepb.Resource {
+	return &resourcepb.Resource{
+		Type:   resourcekeys.ContainerType,
+		Labels: labels,
+	}
+}
+
+// getAllContainerLabels returns all container labels, including ones from
+// the pod in which the container is running.
+func getAllContainerLabels(id string, name string, image string,
```

Comment:
It is easy to mistakenly pass parameters in incorrect order when there are multiple of the same type. Perhaps instead use this:
```suggestion
func getAllContainerLabels(cs ContainerStatus,
```

Reply:
Updated.